### PR TITLE
Add test coverage for RedisEventStore and resque approriately

### LINKED ISSUE #1629: Sufia::RedisEventStore should have a logger
### Descriptive summary

Due to some recent Redis changes, I saw:

``` ruby
Failure/Error: logger.error("unable to create event: #{e}") if logger
NameError:
  undefined local variable or method `logger' for Sufia::RedisEventStore:Class
```

Let's use `CurationConcerns::NullLogger.new` instead.

### Expected behavior

It should just return nil.

### Actual behavior

It doesn't.

### PR DESCRIPTION
**Pull request by awead**
Fixes #1629

`Sufia::RedisEventStore` doesn't have any test coverage, so we're adding some. Let's also resque from the appropriate error and always have some kind of logger.

@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: lib/sufia/redis_event_store.rb [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+        event_id = Redis.current.incr("events:latest_id")
+        Redis.current.hmset("events:#{event_id}", "action", action, "timestamp", timestamp)
+        event_id
+      rescue Redis::CommandError => e
+        logger.error("unable to create event: #{e}")
+        nil
+      end
+
+      def logger
+        @logger = (defined?(Rails) && Rails.logger) || CurationConcerns::NullLogger.new
```
Comment (jcoyne) [inline lib/sufia/redis_event_store.rb]: How could Rails ever be undef here?

Comment (awead) [inline lib/sufia/redis_event_store.rb]: I didn't know the context in which this class was used. For example, if it was used in Resque, but I guess that has Rails as well? Either way, I'm :+1: to removing it.

Comment (jcoyne) [inline lib/sufia/redis_event_store.rb]: Rails is a dependency of Sufia, so it must exist.

Comment (jcoyne) [inline lib/sufia/redis_event_store.rb]: also why are you assigning to `@logger`?

Comment (awead) [inline lib/sufia/redis_event_store.rb]: ah, yeah. habit. it's only called once so no need.

**Comment by mjgiarlo** [comment]
:+1: as soon as the changes to `RedisEventStore` discussed by @jcoyne and @awead are pushed.

**Pull request merged**